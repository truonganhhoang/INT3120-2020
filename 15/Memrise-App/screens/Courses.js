import React, { useState, useEffect } from "react";
import { StyleSheet, View, FlatList, Dimensions } from "react-native";
import * as Progress from "react-native-progress";

import Unit from "../components/Unit";
import axios from "axios";
import Spinner from "../components/Spinner";

const screenWidth = Math.round(Dimensions.get("window").width);

export default function Courses({ navigation }) {
  const [courses, setCourses] = useState({});
  const [isLoading, setIsLoading] = useState(true);
  useEffect(() => {
    const queryString = `http://localhost:3000/courses`;
    axios
      .get(queryString)
      .then((res) => {
        const newCourses = res.data;
        setCourses(newCourses);
        setIsLoading(false);
      })
      .catch((err) => console.log("Error in courseScreen is ", err));
  }, []);

  function onPressNavigate(id) {
    const navigateCourse = courses.filter(
      (course) => course.id === id
    );
      
    return navigation.navigate("ListWord", {
      id: id,
      navigateCourse: navigateCourse,
    });
  }

  return (
    <View style={styles.container}>
      {(isLoading && <Spinner />) || (
        <View style={styles.container}>
          {/* <Progress.Bar progress={0.3} width={screenWidth} /> */}
          <FlatList
            data={courses}
            renderItem={({ item }) => (
              <Unit unit={item} onPress={onPressNavigate} />
            )}
            keyExtractor={(item) => `${item.id}`}
            scrollEnabled={true}
            showsVerticalScrollIndicator={false}
          />
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
    backgroundColor: "#fff",
    alignItems: "stretch",
    justifyContent: "center",
    flex: 1,
  },
});
