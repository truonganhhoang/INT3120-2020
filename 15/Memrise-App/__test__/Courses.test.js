import React from "react";
import renderer from "react-test-renderer";
import Courses from "../screens/Courses";

it("Test courses Screen", () => {
  const tree = renderer.create(<Courses/>).toJSON();
  expect(tree).toMatchSnapshot();
});
