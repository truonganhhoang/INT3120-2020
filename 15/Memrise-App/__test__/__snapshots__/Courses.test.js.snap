// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test courses Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    onLayout={[Function]}
    progress={0.3}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
          "borderColor": "rgba(0, 122, 255, 1)",
          "borderRadius": 4,
          "borderWidth": 1,
          "overflow": "hidden",
          "width": 750,
        },
        undefined,
      ]
    }
    useNativeDriver={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 122, 255, 1)",
          "height": 6,
          "transform": Array [
            Object {
              "translateX": -2.842170943040401e-14,
            },
            Object {
              "translateX": -261.8,
            },
            Object {
              "scaleX": 0.30007,
            },
          ],
        }
      }
    />
  </View>
  <RCTScrollView
    data={Object {}}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
